### PR TITLE
refactor: migrage Bitmap from arrow2 to arrow-rs

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16523)

### CODE REVIEW

File: src/query/expression/src/types/boolean.rs [hunk 1]
```diff
@@ -36,10 +37,10 @@ pub struct BooleanType;
 impl ValueType for BooleanType {
     type Scalar = bool;
     type ScalarRef<'a> = bool;
-    type Column = Bitmap;
+    type Column = BooleanBuffer;
```
Comment (sundy-li) [inline src/query/expression/src/types/boolean.rs:40]: Why not use `NullBuffer`, it has materialized null_count.

Comment (andylokandy) [reply src/query/expression/src/types/boolean.rs:40]: I can not find a builder for `NullBuffer` and it does not `impl BitAnd`

Comment (sundy-li) [reply src/query/expression/src/types/boolean.rs:40]: 1. `NullBuffer::new()`
2. `pub fn union(lhs: Option<&NullBuffer>, rhs: Option<&NullBuffer>) -> Option<NullBuffer> `

Comment (sundy-li) [reply src/query/expression/src/types/boolean.rs:40]: Also: 
`impl From<BooleanBuffer> for NullBuffer`

Comment (andylokandy) [reply src/query/expression/src/types/boolean.rs:40]: But these all operation is direct on `BooleanBuffer`: it implments BitAnd, it provides `value_unchecked`, and have `BooleanBufferBuilder`. I think `BoolteanBuffer` is the directly replacement for `Bitmap` in arrow2.

**Comment by Dousir9** [comment]
Please write a rust benchmark to test the migrated functions in kernels(`take.rs, concat.rs, ...`) to ensure that there is no performance regression.

**Pull request closed without merging**